dataprep/clean/clean_headers.py

```diff
@@ -0,0 +1,273 @@
+"""
+Implement clean_headers function
+"""
+import re
+from typing import Any, Optional, Union, List, Dict
+from unicodedata import normalize
+
+import dask
+import dask.dataframe as dd
+import pandas as pd
+
+from ..eda.progress_bar import ProgressBar
+from .utils import NULL_VALUES, to_dask
+
+
+def clean_headers(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    case: str = "snake",
+    replace: Optional[Dict[str, str]] = None,
+    remove_accents: bool = True,
+    report: bool = True,
+    progress: bool = True,
+) -> pd.DataFrame:
+    """
+    Function to clean headers / column names.
+
+    Parameters
+    ----------
+    df
+        Dataframe from which column names are to be cleaned.
+
+    case
+        The desired case style of the column name.
+            - 'snake': 'column_name'
+            - 'kebab': 'column-name'
+            - 'camel': 'columnName'
+            - 'pascal': 'ColumnName'
+            - 'const': 'COLUMN_NAME'
+            - 'sentence': 'Column name'
+            - 'title': 'Column Name'
+            - 'lower': 'column name'
+            - 'upper': 'COLUMN NAME'
+
+        (default: 'snake')
+
+    replace
+        Values to replace in the column names.
+            - {"old_value": "new_value"}
+
+        (default: None)
+
+    remove_accents
+        If True, strip accents from the column names.
+
+        (default: True)
+
+    report
+        If True, output the summary report. Otherwise, no report is outputted.
+
+        (default: True)
+
+    progress
+        If True, enable the progress bar.
+
+        (default: True)
+
+    Examples
+    --------
+    Clean column names by converting the names to camel case style, removing accents,
+    and correcting a mispelling.
+
+    >>> df = pd.DataFrame({"FirstNom":["Philip", "Turanga"], "lastName": ["Fry", "Leela"],
+                           "Téléphone":["555-234-5678", "(604) 111-2335"]})
+    >>> clean_headers(df, case='camel', replace={"Nom": "Name"})
+    Headers Cleaning Report:
+        2 values cleaned (66.67%)
+    Result contains 3 (100.0%) values in the correct format and 0 null values (0.0%)
+      firstName lastName       telephone
+    0    Philip      Fry    555-234-5678
+    1   Turanga    Leela  (604) 111-2335
+    """
+    # pylint: disable=too-many-arguments
+
+    case_styles = {
+        "snake",
+        "kebab",
+        "camel",
+        "pascal",
+        "const",
+        "sentence",
+        "title",
+        "lower",
+        "upper",
+    }
+
+    if case not in case_styles:
+        raise ValueError(
+            f"case {case} is invalid, it needs to be one of {', '.join(c for c in case_styles)}"
+        )
+
+    # Convert to dask
+    df = to_dask(df)
```

Comment:
I recommend not going converting to dask. I think the API to access the columns for a Dask and pandas DataFrame is `df.columns`, so I don't think you will need to consider whether a Dask or pandas df is passed.